test/test_rake_file_utils.rb

```diff
@@ -287,6 +287,110 @@ def test_sh_show_command
     assert_equal expected_cmd, show_cmd
   end
 
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_echoes_it_fully_long
+    assert_sh_echoes_it_fully(*string_and_output_and_error_long)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_echoes_it_fully_short
+    assert_sh_echoes_it_fully(*string_and_output_and_error_short)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_prints_its_full_output_when_verbose_long
+    verbose true
+    assert_sh_prints_its_full_output(*string_and_output_and_error_long)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_prints_its_full_output_when_verbose_short
+    verbose true
+    assert_sh_prints_its_full_output(*string_and_output_and_error_short)
```

Comment:
I'm having a really hard time figuring out what these tests are checking.

I don't think reduced duplication will help me maintain these tests.

Can you expand these so that I can see some setup code, `sh` invocation, then some assertion for each test?